### PR TITLE
Fogl 475

### PR DESCRIPTION
**Pull request by vaibhav-ScaleDB**
### TODO / DISCUSS:

- FOGL-482: Confirm endpoint for DELETE  is `http://localhost:8082/foglamp/category/{category_name}/{config_item}/value` [As per [GDoc](https://docs.google.com/document/d/1JJDP7g25SWerNVCxgff02qp9msHbqA9nt3RAFx8-Qng/edit#heading=h.nlvnmq65bwpa)] OR `http://localhost:8082/foglamp/category/{category_name}/{config_item}` [As per current implementation]
- FOGL-481: Endpoint not defined for adding (merging) a new config item to existing config?

### CODE REVIEW

**Comment by vaibhav-ScaleDB** [comment]
@ashish-ScaleDB , @amarendra-dianomic Please address the TODO comments for configuration manager restapi

**Comment by ashish-ScaleDB** [comment]
#1 - DELETE verb is obviously to be used as to delete some data or resource, but  PUT is the right verb in case, @praveen-garg has also commented on it. I would suggest to go ahead with DELETE for now.
#2 AFAIK we can do only set and unset config item value ONLY

File: src/python/tests/core/api/test_configuration.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                       test_data['key'], test_data['description'], json.dumps(test_data['value']))
+    await conn.close()
+
+
+async def delete_master_data():
+    conn = await asyncpg.connect(database=__DB_NAME)
+    await conn.execute('''DELETE from foglamp.configuration WHERE key IN ($1)''', test_data['key'])
+    await conn.close()
+
+
```
Comment (ashish-ScaleDB) [inline src/python/tests/core/api/test_configuration.py]: Please add allure feature and story annotation for Jenkins allure-report
```
@pytest.allure.feature("api")
@pytest.allure.story("configuration")
```

File: src/python/tests/core/api/test_configuration.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        conn.close()
+        retval = json.loads(r)
+        test_data_item_value.update(body)
+        assert test_data_item_value == retval
+
+    async def test_unset_config_item(self):
+        conn = http.client.HTTPConnection(BASE_URL)
+        test_data_item = [key for key in test_data['value']][0]
+
+        # TODO: FOGL-482 endpoint is DELETE http://localhost:8082/foglamp/category/{category_name}/{config_item}/value
```
Comment (ashish-ScaleDB) [inline src/python/tests/core/api/test_configuration.py]: FOGL-482 has already done, please update accordingly.

**Pull request merged**